src/components/HeaderBar.vue

```diff
@@ -0,0 +1,96 @@
+<script setup>
+import IconBaseball from '../components/icons/IconBaseball.vue';
+import IconBasketball from '../components/icons/IconBasketball.vue';
+import IconFootball from '../components/icons/IconFootball.vue';
+import IconHockey from '../components/icons/IconHockey.vue';
+import { ref } from 'vue';
+
+const props = defineProps({ allSports: { type: Array, required: true } })
+// let thisComponent = ref(IconBaseball)
+// ^^^tried using this to make the component :is dynamic, but will circle back to it
+</script>
+
+<template>
+  <div class="container">
+    <div class="sports-wrapper">
+      <button
+        v-for="sport in allSports" 
+        :key="sport.key"
+        class="sport"
+      >
+        <IconBaseball
+          v-if="sport.key === 'mlb'"
+          class="icon"
+        />
+      
+        <IconHockey
+          v-else-if="sport.key === 'nhl'" 
+          class="icon"
+        />
+      
+        <IconBasketball
+          v-else-if="sport.key === 'nba'" 
+          class="icon"
+        />
+      
+        <IconFootball
+          v-else-if="sport.key === 'nfl'" 
+          class="icon"
+        />
+      
+        <p class="button-text">
+          {{ sport.key.toUpperCase() }}
+        </p>
+      </button>
+    </div>
+  </div>
+</template>
+
+<style>
+.container {
+  width: 100%;
+}
+
+.sports-wrapper {
+  height: 50px;
+  width: 100%;
+  display: flex;
+  justify-content: center;
+  align-items: center;
+}
+
+.sport {
+  display: flex;
+  justify-content: center;
+  align-items: center;
+  transform: skew(-30deg);
+  padding-left: 6VW;
+  padding-right: 6VW;
+  outline: none;
+  border: none;
+}
+
+button{
+  height: 64px;
+  transition: background-color ease-in-out .3s;
+  background-color: #101c33;
+  color: #a7a7a7;
+}
+
+button:hover {
+  transition: background-color ease-in-out .3;
+  background-color: #2b3648;
+  color: #ffff;
+}
```

Comment:
Leverage `sass` when you can. So you can nest `:hover` and other selectors in it's parent just by using `&`:

```
button {
   &:hover {

   }

   &:disabled {

    }
```

etc.

Reply:
Cool! 

is sass already installed? I looked in the package.json and didnt see it, but maybe that's not even where I would look to check 🤷‍♂️

will google

Reply:
Just added it, make sure to fetch and pull, and then `yarn install`